plugin/v2/settable.go

```diff
@@ -0,0 +1,89 @@
+package v2
+
+import (
+	"fmt"
+	"strings"
+)
+
+type settable struct {
+	name  string
+	field string
+	value string
+}
+
+var (
+	allowedSettableFieldsEnv     = []string{"value"}
+	allowedSettableFieldsArgs    = []string{"value"}
+	allowedSettableFieldsDevices = []string{"path"}
+	allowedSettableFieldsMounts  = []string{"source"}
+)
+
+func newSettables(args []string) []settable {
+	settables := []settable{}
+	for _, arg := range args {
+		settables = append(settables, newSettable(arg))
+	}
+	return settables
+}
+
+func newSettable(arg string) settable {
+	var set settable
+	if i := strings.Index(arg, "="); i < 0 {
+		set.name = arg
+		set.value = "true"
+	} else {
+		set.name = arg[:i]
+		set.value = arg[i+1:]
+	}
+
+	if i := strings.LastIndex(set.name, "."); i > 0 {
+		set.field = set.name[i+1:]
+		set.name = arg[:i]
+	}
+
+	return set
+}
+
+// prettyName return name.field if there is a field, otherwise name.
+func (set *settable) prettyName() string {
+	if set.field != "" {
+		return fmt.Sprintf("%s.%s", set.name, set.field)
+	}
+	return set.name
+}
+
+func (set *settable) isSettable(allowedSettableFields []string, settable []string) bool {
+	if set.field == "" && len(settable) > 0 {
+		// assume field = settable[0]
```

Comment:
@vieux not sure if it's just the comment that's incorrect or the implementation as well but I believe we agreed that `settable[0]` is the default, only if `len(settable) == 1` not if `len(settable) > 1`.